core/Tracker/Visit.php

```diff
@@ -158,6 +158,14 @@ public function handle()
 
         $isNewVisit = $this->isVisitNew($visitor, $action);
 
+        // on a ping request that is received before the standard visit length, we just update the visit time w/o
+        // adding a new action
+        if ($this->isPingRequest()
```

Comment:
can be maybe put into the `if` of link `176` where it already checks for `$isNewVisit`? Do we still have to execute all the following `$this->goalManager->*` logic in such a case? I'm not so much into it right now. Maybe it would be faster but probably adds lots of complexity. I might have another look tmrw to understand it better